### DISCUSSION TITLE
How does server side cookie parsing work on the `with-firebase-authentication` example?

### DISCUSSION THREAD

**Discussion by devth** [vercel/next.js#15057] [Help] [answered]
I read through the source of [with-firebase-authentication](https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication). It uses `js-cookie` to read and write cookies. However, when I try this in my app cookies are always `undefined` on the server side. I can see the cookie in in `req.headers.cookie` server side, but `js-cookie` does not magically obtain that value. I don't see `wire-firebase-authentication` parsing the cookies anywhere so I don't understand how it could actually work. Am I missing something?

Thanks 🙏

**Comment by leerob** [comment, marked as answer]
`js-cookie` is for handling _browser_ cookies, not on the server-side. They pass the token as a header to the backend. It looks like they parse [`req.headers.token`](https://github.com/vercel/next.js/blob/canary/examples/with-firebase-authentication/pages/api/getFood.js#L5), which comes from [here](https://github.com/vercel/next.js/blob/canary/examples/with-firebase-authentication/components/FirebaseAuth.js#L27).

**Reply by devth** [reply]
Thanks @leerob. I was able to solve my issue using [next-cookies](https://www.npmjs.com/package/next-cookies) for server-side.

**Reply by francisrod01** [reply]
@leerob @devth I have a question about it.
Does the example need the `next-cookies`  to read/update `redux`/`redux-persist` state on server-side?
